### ISSUE TITLE
Empty list of instances

### ISSUE THREAD

**Issue by abelgomez** [turtle0x1/LxdMosaic#262]
**Describe the bug**
It turns out that in some host configurations, the list of instances in a host may be rendered as empty due to missing properties on the platform because lxdMosaic tries to navigate `null` properties near lines: https://github.com/turtle0x1/LxdMosaic/blob/0691b003b358c24d0634e92f03eed89d105a38c4/src/views/boxes/server.php#L378-L402.

I didnt' investigate where these values come from, but this is the case for LXD/lxdMosaic installed on a Raspberry PI. The properties that are null in this platform are: `data.resources.system.motherboard`, `data.resources.system.chassis` and `data.resources.system.firmware`.

A `null` check should be done (or the upcoming [optional chaining operator ](https://github.com/tc39/proposal-optional-chaining/) may be used intead in the future, e.g., `data?.resources?.system?.motherboard?.product`).

**To Reproduce**
1. Go to the dashboard of a host deployed on a platform with missing properties.

**Expected behavior**
A list of instances shoud be shown (e.g., in the screenshot the lxdMosaic instance is missing from the dashboard).

**Screenshots**
![imagen](https://user-images.githubusercontent.com/5593904/94633211-ee540600-02cc-11eb-914f-876b8741339b.png)

**Desktop (please complete the following information):**
 - OS: Raspbian Buster
 - Browser: Firefox and Chrome
 - Version: 0.9.3 (and HEAD)

**Comment by turtle0x1** [comment]
Thank you for your thorough investigation the commit above should resolve the issue (I opted to go with explicit checking so I could do "Not Set" instead of using `?.` which would just should `null`, thank you for suggesting it though :smile:)

**Comment by abelgomez** [comment]
Great, thanks. I tested HEAD, and I can confirm the issue is solved.

Your null check is more portable (and conformant to the current standard) so it's perfect. I just suggested `?.` because it's easier and more concise (but, BTW, the "elvis" operator shows `undefined` rather than `null` 🙂).

Just one question: do you know when you're going to release v0.10.0?

There are a couple of details to be fixed in the installation script in HEAD for Raspbian.

**Comment by turtle0x1** [comment]
`v0.10.0` will probably be in the next couple of weeks, its getting quite thick with changes already so ill probably have to bump some the issues out of it :(

I have just moved so all my RPI's are boxed up right now, if you could submit a PR id be grateful :smile:

**Comment by abelgomez** [comment]
Sure! The problems I faced came basically with node packages that needed to be compiled from sources. I thik that it's not worth to have this condition in the script since there may be more archs that need to build from sources:

https://github.com/turtle0x1/LxdMosaic/blob/04d2a4aa62c8e7f0cda6f9d1e0980369d9b36615/examples/install_with_clone.sh#L38-L41

For Raspbian I opted for installing `build-essential` directly prior to install lxdMosaic

Also, from a quick inspection, the inclusion of sqlite3 in this commit https://github.com/turtle0x1/LxdMosaic/commit/71d26d8a69409723f652479e87318ca8c1e1b302 makes the installation fail, since there are no prebuilt packages for armhf and then also python2 is also required for building sqlite3 from sources.

Anyway, I'll test and open an issue if needed (with a PR) when v.0.10.0 is released.

Thanks!
